protoc_plugin/lib/src/shared.dart

```diff
@@ -2,8 +2,61 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:convert';
+
+import '../protoc.dart';
+
 const protobufImportPrefix = r'$pb';
 const asyncImportPrefix = r'$async';
 const coreImportPrefix = r'$core';
 const grpcImportPrefix = r'$grpc';
 const mixinImportPrefix = r'$mixin';
+
+extension FileDescriptorProtoExt on FileGenerator {
+  String? commentBlock(List<int> path) {
+    if (!options.generateComments) {
+      return null;
+    }
+
+    final bits = descriptor.sourceCodeInfo.location
+        .where((element) => element.path.toString() == path.toString())
+        .toList();
+
+    if (bits.length == 1) {
+      final match = bits.single;
+      if (match.hasLeadingComments()) {
+        return toDartComment(match.leadingComments);
+      }
```

Comment:
```suggestion
      return toDartComment(match.leadingComments);
```

When the field `leadingComments` isn't present the field value will be the default value of the field which is empty string, and `toDartComment` handles empty strings. So no need to check for presence.